### PR TITLE
SOFT 170 can_handler for bms carrier

### PR DESCRIPTION
**Pull request by shahhet14**
Created can_handler module to tx current, voltage, temperature, relay states and fan states.

### CODE REVIEW

File: projects/bms_carrier/src/can_handler.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  } else if (s_msgs_txed < NUM_TOTAL_MESSAGES) {
+    prv_fan_status_tx(storage);
+  }
+}
+
+StatusCode can_handler_init(BmsStorage *storage) {
+  if (storage == NULL) {
+    return STATUS_CODE_INVALID_ARGS;
+  }
+  prv_periodic_tx(SOFT_TIMER_INVALID_TIMER, storage);
```
Comment (g-jessmuir) [inline projects/bms_carrier/src/can_handler.c:75]: I think it makes sense to actually start a soft timer here for 1 second to give other things a chance to collect readings before TXing them. You can either set the 1 second to a #define or just add a comment explaining it's a one-off delay to let data accumulate

**Review by g-jessmuir — CHANGES_REQUESTED** [review]
just a few small changes then we should be good, thanks! also move your codegen tooling PR out of draft if you're good to merge

File: libraries/codegen-tooling/inc/can_pack.h [hunk 2]
```diff
@@ -241,23 +241,24 @@
...
+                   8, (fan_1_u8), (fan_2_u8), (fan_3_u8), (fan_4_u8), (fan_5_u8), (fan_6_u8),      \
+                   (fan_7_u8), (fan_8_u8))
 
 #define CAN_PACK_BATTERY_RELAY_STATE(msg_ptr, hv_u8, gnd_u8)                      \
   can_pack_impl_u8((msg_ptr), SYSTEM_CAN_DEVICE_BMS_CARRIER,                      \
                    SYSTEM_CAN_MESSAGE_BATTERY_RELAY_STATE, 2, (hv_u8), (gnd_u8),  \
                    CAN_PACK_IMPL_EMPTY, CAN_PACK_IMPL_EMPTY, CAN_PACK_IMPL_EMPTY, \
                    CAN_PACK_IMPL_EMPTY, CAN_PACK_IMPL_EMPTY, CAN_PACK_IMPL_EMPTY)
 
+#define CAN_PACK_SOLAR_DATA(msg_ptr, data_point_type_u32, data_value_u32)                 \
```
[the remark this answers is not in the document]
Comment (shahhet14) [reply libraries/codegen-tooling/inc/can_pack.h:256]: Not intentional, the only changes should've been the changes to the fan state message. Might be an outdated branch or something.

**Comment by shahhet14** [comment]
Made the changes you requested. For the wait between messages, I've made it 85 ms for now.

**Review by g-jessmuir — APPROVED** [review]

**Pull request merged**